var/spack/repos/builtin/packages/py-awkward/package.py

```diff
@@ -31,13 +43,19 @@ class PyAwkward(PythonPackage):
     patch("pybind11.patch", when="@:1.2.2")
     patch("pybind11_02.patch", when="@1.2.3:1.8.0")
 
-    depends_on("py-setuptools@42.0:", type=("build", "run"))
+    depends_on("py-setuptools@42.0:", when="@:1", type=("build", "run"))
     depends_on("py-pyyaml", type="build")
+    depends_on("py-hatchling@1.10:", when="@2:", type="build")
+    depends_on("py-hatch-fancy-pypi-readme", when="@2:", type="build")
 
     depends_on("python@2.7:2.8,3.5:", type=("build", "run"))
-    depends_on("python@3.6:", when="@1.9.0:", type=("build", "run"))
-    depends_on("py-numpy@1.13.1:", type=("build", "run"))
+    depends_on("python@3.6:", when="@1.9:", type=("build", "run"))
+    depends_on("python@3.7:", when="@1.10:", type=("build", "run"))
+    depends_on("py-numpy@1.13.1:", when="@:1", type=("build", "run"))
+    depends_on("py-numpy@1.14.5:", when="@2:", type=("build", "run"))
     depends_on("py-pybind11", type=("build", "link"))
+    depends_on("py-importlib-resources", when="@2: ^python@:3.8", type=("build", "run"))
+    depends_on("py-typing-extensions@4.1:", when="@2: ^python@:3.10", type=("build", "run"))
```

Comment:
Also missing dep on packaging.

Reply:
How does one tell from the pyproject.toml file what is needed for build vs run? Presumably packaging is just a build dependency.
```suggestion
    depends_on("py-typing-extensions@4.1:", when="@2: ^python@:3.10", type=("build", "run"))
    depends_on("py-packaging", when="@2:", type=("build", "run"))
```

Reply:
https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#dependencies

Reply:
If you grep for imports you can also see that it's needed at run-time. But I usually only look at pyproject.toml/setup.cfg/setup.py and hope that the authors know what they're doing.